### DISCUSSION TITLE
How to add a "Date" field in a nested object of a Single View?

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#341] [Q&A] [answered]
### Which section?

Fast Data

### Which is the version?

11.7.1

### Description

Hi!

I need to configure a Single View which contains an "orders" field of type "array of object". Each object has a field "orderDate" of type "Date". While for root-level fields of a Single View you can configure the type "Date" from Console, for nested objects you cannot because JSON schema does not allow any "Date" type.

Is there a way to have "Date" types in nested objects?

Thanks!

**Comment by ThisIsDemetrio** [comment, marked as answer]
Hi, @Lucascanna!

Regarding the configuration of fields inside the Single View, while top level fields uses a custom notation (which allows to use the Date type), fields inside object and array of objects uses the [default JSON Schema notation](https://json-schema.org/).

So, you have to set the type of your field to _string_, then add the _format_ you prefer according to the [available types](https://json-schema.org/understanding-json-schema/reference/string#dates-and-times). Something like this:
```
"orderDate": {
  "type": "string",
  "format": "date-time"
}
```

Hope this helps!
